### PR TITLE
Use larger timeout for all Http calls

### PR DESCRIPTION
**Pull request by EmileSonneveld**
https://github.com/Open-EO/openeo-geopyspark-driver/issues/342

### CODE REVIEW

**Comment by EmileSonneveld** [comment]
I'll check again tomorrow about the RESTO API. Maybe it is working again by then.

File: src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala [hunk 1]
```diff
@@ -48,6 +48,7 @@ object CreodiasClient extends OpenSearchClient {
       .param("status", "0|34|37")
       .param("dataset", "ESA-DATASET")
       .params(attributeValues.mapValues(_.toString).filterKeys(!Seq( "eo:cloud_cover", "provider:backend", "orbitDirection", "sat:orbit_state").contains(_)).toSeq)
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala:51]: This is already covered by the change in OpenSearchClient.

File: src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala [hunk 2]
```diff
@@ -76,6 +77,7 @@ object CreodiasClient extends OpenSearchClient {
   override def getCollections(correlationId: String): Seq[Feature] = {
     val getCollections = http(collections)
       .option(HttpOptions.followRedirects(true))
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala:80]: This is already covered by the change in OpenSearchClient.

File: src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala [hunk 3]
```diff
@@ -20,6 +20,7 @@ class OscarsClient(val endpoint: URL, val isUTM:Boolean = false) extends OpenSea
         .param("sortKeys", key)
         .param("count", "1")
         .params(attributeValues.mapValues(_.toString).toSeq)
+        .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala:23]: I would delegate to superclass method org.openeo.opensearch.OpenSearchClient#http instead.

File: src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala [hunk 4]
```diff
@@ -66,6 +67,7 @@ class OscarsClient(val endpoint: URL, val isUTM:Boolean = false) extends OpenSea
       .param("startIndex", page.toString)
       .params(newAttributeValues.mapValues(_.toString).filterKeys(!Seq( "eo:cloud_cover", "provider:backend").contains(_)).toSeq)
       .param("clientId", clientId(correlationId))
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala:70]: This is already covered by the change in OpenSearchClient.

File: src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala [hunk 5]
```diff
@@ -95,6 +97,7 @@ class OscarsClient(val endpoint: URL, val isUTM:Boolean = false) extends OpenSea
     val getCollections = http(s"$endpoint/collections")
       .option(HttpOptions.followRedirects(true))
       .param("clientId", clientId(correlationId))
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala:100]: This is already covered by the change in OpenSearchClient.

File: src/main/scala/org/openeo/opensearch/backends/STACClient.scala [hunk 6]
```diff
@@ -53,6 +53,7 @@ class STACClient(private val endpoint: URL=new URL("https://earth-search.aws.ele
       .param("limit", "100")
       .param("bbox", bboxParam)
       .param("page", page.toString)
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/STACClient.scala:56]: This is already covered by the change in OpenSearchClient.

File: src/main/scala/org/openeo/opensearch/backends/STACClient.scala [hunk 7]
```diff
@@ -69,6 +70,7 @@ class STACClient(private val endpoint: URL=new URL("https://earth-search.aws.ele
   override def getCollections(correlationId: String = ""): Seq[Feature] = {
     val getCollections = http(s"$endpoint/collections")
       .option(HttpOptions.followRedirects(true))
+      .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```
Comment (bossie) [inline src/main/scala/org/openeo/opensearch/backends/STACClient.scala:73]: This is already covered by the change in OpenSearchClient.

**Review by bossie — CHANGES_REQUESTED** [review]
Has some redundant calls.

File: src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala (hunk 1 above)
Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala]: fixed

File: src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala (hunk 2 above)
Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala]: fixed

File: src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala (hunk 3 above)
Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala]: Good idea. As a bonus, redirects can be followed too here.

File: src/main/scala/org/openeo/opensearch/backends/STACClient.scala (hunk 6 above)
Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/backends/STACClient.scala]: fixed

File: src/main/scala/org/openeo/opensearch/backends/STACClient.scala (hunk 7 above)
Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/backends/STACClient.scala]: fixed

**Review by bossie — APPROVED** [review]
Looks good.

**Pull request merged**